examples/Protobuf/messages/CMakeLists.txt

```diff
@@ -1,7 +1,41 @@
+# Example for how to compile .proto files into a library
+# Both the 'Module Compatible' and new way are shown
 
-# required since no linking
-include_directories(${PROTOBUF_INCLUDE_DIR})
-PROTOBUF_GENERATE_CPP(MESSAGES_SRC MESSAGES_HEADER messages.proto)
-add_library(messages ${MESSAGES_HEADER} ${MESSAGES_SRC})
+set(MESSAGES_PROTOS messages.proto)
+set(MESSAGES_SRCS)
+set(MESSAGES_HDRS)
 
+#Code Generation
+if(protobuf_MODULE_COMPATIBLE) #Legacy Support
```

Comment:
Is there anything behind the -imported postfix? 

So should i name the ${PROTOBUF_INCLUDE_DIR} Version Protobuf
and the protobuf::libprotobuf Version Protobuf-imported?